lib/request.js

```diff
@@ -249,7 +250,7 @@ Test.prototype.then = function (onResolve, onReject) {
     var self = this;
     this._promise = new Test.Promise(function (resolve, reject) {
       self.end(function (err, res) {
-        if (err) {
+        if (err && self.options.badStatusCausesError) {
```

Comment:
Not sure I like this code right here - because bad statuses aren't the only cause of failure (for example timeouts, dns failures, connection closures all produce errors) - some of those errors are genuinely errors and should always fail. This should check the `res.status` as well.